### ISSUE TITLE
TypeError: Cannot read property 'selections' of undefined

### ISSUE THREAD

**Issue by Banhawy** [neo4j-graphql/neo4j-graphql-js#145]
I have the following GraphQL schema:
```
type User {
    fName: String!
    lName: String!
    socialAccounts: SocialMedia @relation(name: "HAS", direction: "OUT")
}

input UserInput {
    fName: String!
    lName: String!
    SocialMedia: SocialMediaInput
}

type SocialMedia {
    facebook: Facebook
}

input SocialMediaInput {
    facebook: FacebookInput
}

type Facebook {
    url: String
}

input FacebookInput {
    url: String
}

type Mutation {
    UpdateSocialMedia(user: UserInput): String @cypher(
        statement: "MERGE (u:User) WHERE u.lName = $user.lName MERGE (m:Facebook) WHERE m.url = $user.SocialMedia.facebook.url MERGE (u)-[:HAS]->(m) return m"
    )
}
```
I am trying to create a mutation that can update a user's social media url where that social media type is a labeled node in neo4j. The problem I get is that the GraphQL mutation doesn't execute. The following mutation:
```
UpdateSocialMedia(
    user: {
      fName: "Adham",
      lName: "El Banhawy",
      SocialMedia: {
        facebook: {
          url: "https://www.facebook.com/benhawy"
        }
      }
    }
  )
```
gives me the following error stack:
```
"errors": [
    {
      "message": "Cannot read property 'selections' of undefined",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "UpdateSocialMedia"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: Cannot read property 'selections' of undefined",
            "    at cypherMutation (A:\\Coding\\projects\\ProjectX\\api\\node_modules\\neo4j-graphql-js\\dist\\index.js:244:84)",
            "    at _callee$ (A:\\Coding\\projects\\ProjectX\\api\\node_modules\\neo4j-graphql-js\\dist\\index.js:65:31)",
            "    at tryCatch (A:\\Coding\\projects\\ProjectX\\api\\node_modules\\babel-runtime\\node_modules\\regenerator-runtime\\runtime.js:62:40)",
            "    at Generator.invoke [as _invoke] (A:\\Coding\\projects\\ProjectX\\api\\node_modules\\babel-runtime\\node_modules\\regenerator-runtime\\runtime.js:296:22)",
            "    at Generator.prototype.(anonymous function) [as next] (A:\\Coding\\projects\\ProjectX\\api\\node_modules\\babel-runtime\\node_modules\\regenerator-runtime\\runtime.js:114:21)",
            "    at step (A:\\Coding\\projects\\ProjectX\\api\\node_modules\\babel-runtime\\helpers\\asyncToGenerator.js:17:30)",
            "    at A:\\Coding\\projects\\ProjectX\\api\\node_modules\\babel-runtime\\helpers\\asyncToGenerator.js:35:14",
            "    at Promise (<anonymous>)",
            "    at F (A:\\Coding\\projects\\ProjectX\\api\\node_modules\\core-js\\library\\modules\\_export.js:36:28)",
            "    at A:\\Coding\\projects\\ProjectX\\api\\node_modules\\babel-runtime\\helpers\\asyncToGenerator.js:14:12"
          ]
        }
      }
    }
  ],
```
Is there a way to get this working to match the GraphQL mutation format?

Neo4j Version: 3.4.1 Enterprise 
neo4j-graphql Version:  3.4.0.1

**Comment by AbbesBendaoud10** [comment]
Hello , can you give us the code of your back end api ? in where you consume the mutation please  ?

**Comment by AbbesBendaoud10** [comment]
Can you send me screen shot of your GraphQL Playground/editor reponse ?

**Comment by Banhawy** [comment]
Here's the screenshot:
![screenshot-localhost-4040-2018 11 22-20-40-48](https://user-images.githubusercontent.com/9264853/48918784-7b3f0480-ee97-11e8-85ef-88d4a51cf35f.png)

**Comment by johnymontana** [comment]
@Banhawy It seems we assume mutations will return object types, and not scalars. As a workaround can you try returning the `User` in your mutation:

```
type Mutation {
    UpdateSocialMedia(user: UserInput): User @cypher(
        statement: "MERGE (u:User {lName: $user.lName}) MERGE (m:Facebook {url: $user.SocialMedia.facebook.url}) MERGE (u)-[:HAS]->(m) return u"
    )
}
```

NOTE: I also changed your Cypher query a bit - `WHERE` can't be used with `MERGE`, instead use the inline curly braces.

**Comment by btroop** [comment]
I seem to be running into a similar issue, but on the Query type:

```
type User {
  uuid: ID
  firstName: String
  lastName: String
}

type Query {
 userCount: Int @cypher(statement:"Match (:User) RETURN COUNT(*)")
}
```

resulting in the following playground response:

```
{
  "errors": [
    {
      "message": "Cannot read property 'selections' of undefined",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "userCount"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: Cannot read property 'selections' of undefined",
            "    at getQuerySelections (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-graphql-js/dist/utils.js:915:63)",
            "    at cypherQuery (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-graphql-js/dist/index.js:116:50)",
            "    at _callee$ (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-graphql-js/dist/index.js:45:31)",
            "    at tryCatch (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/regenerator-runtime/runtime.js:62:40)",
            "    at Generator.invoke [as _invoke] (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/regenerator-runtime/runtime.js:296:22)",
            "    at Generator.prototype.(anonymous function) [as next] (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/regenerator-runtime/runtime.js:114:21)",
            "    at step (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)",
            "    at /Users/ben/Application Development/ag-app-grand-stack/api/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14",
            "    at new Promise (<anonymous>)",
            "    at new F (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/core-js/library/modules/_export.js:36:28)"
          ]
        }
      }
    }
  ],
  "data": {
    "userCount": null
  }
}
```


I tried adjusting the query to return an array of the User types:

```
type Query {
  userCount: [User] @cypher(statement:"Match (u:User) RETURN u")
}
```

which creates the following playground error:

```
{
  "errors": [
    {
      "message": "Invalid input ',': expected whitespace, comment or an expression (line 1, column 60 (offset: 59))\n\"WITH apoc.cypher.runFirstColumn(\"Match (u:User) RETURN u\", , True) AS x UNWIND x AS `user`\"\n                                                            ^",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "userCount"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "code": "Neo.ClientError.Statement.SyntaxError",
          "name": "Neo4jError",
          "stacktrace": [
            "Neo4jError: Invalid input ',': expected whitespace, comment or an expression (line 1, column 60 (offset: 59))",
            "\"WITH apoc.cypher.runFirstColumn(\"Match (u:User) RETURN u\", , True) AS x UNWIND x AS `user`\"",
            "                                                            ^",
            "",
            "    at captureStacktrace (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-driver/lib/v1/result.js:200:15)",
            "    at new Result (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-driver/lib/v1/result.js:73:19)",
            "    at Session._run (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-driver/lib/v1/session.js:173:14)",
            "    at Session.run (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-driver/lib/v1/session.js:154:19)",
            "    at _callee$ (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/neo4j-graphql-js/dist/index.js:60:28)",
            "    at tryCatch (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/regenerator-runtime/runtime.js:62:40)",
            "    at Generator.invoke [as _invoke] (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/regenerator-runtime/runtime.js:296:22)",
            "    at Generator.prototype.(anonymous function) [as next] (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/regenerator-runtime/runtime.js:114:21)",
            "    at step (/Users/ben/Application Development/ag-app-grand-stack/api/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)",
            "    at /Users/ben/Application Development/ag-app-grand-stack/api/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14"
          ]
        }
      }
    }
  ],
  "data": {
    "userCount": null
  }
}
```

I've tried testing these in Neo4j desktop (I have the graphql plugin installed) and the queries seem work fine from that perspective.

Re: the second error, I'm wondering if it is related to the `WITH apoc.cypher.runFirstColumn()` procedure returning `'user'` as a string rather than `user` as a neo4j variable. 

My environment is set up nearly identically as @Banhawy  (src.js files) except that I'm running 3.5.

Thoughts/help appreciated.

**Comment by johnymontana** [comment]
Another example of this issue: https://community.neo4j.com/t/neo4j-grapql-js-custom-cypher-query/5017

**Comment by johnymontana** [comment]
Fixed in #200
